uint16 curr_spot
string spot_options
string wpts